testing/rspec/spec/unit/linters/unique_scenario_names_linter_spec.rb

```diff
@@ -0,0 +1,261 @@
+            | Something      |
+        """
+      end
+
+      let(:model) do
+        generate_feature_model(source_text: feature_file, parent_file_path: 'path_to_file').tests
+      end
+
+      it 'returns a detected problem and its location' do
+        results = model.map { |scenario| linter.lint(scenario) }.compact
+        expect(results).to include(
+          {
+            problem:  'Scenario names created by Scenario Outline are not unique',
+            location: 'path_to_file:4'
+          }
+        )
+      end
+    end
+
+    context 'with an outline generated scenario name is identical to a regular scenario' do
+      let(:feature_file) do
+        """
+        Feature: Feature with Mixed Scenarios
+
+          Scenario: Duplicate Scenario Name
+            Given something
+
+          Scenario Outline: Duplicate Scenario Name
+            Given something
+
+          Examples:
+            | input          |
+            | Something      |
+        """
+      end
+
+      let(:model) do
+        generate_feature_model(source_text: feature_file, parent_file_path: 'path_to_file').tests
+      end
+
+      it 'returns a detected problem and its location' do
+        results = model.map { |scenario| linter.lint(scenario) }.compact
+        expect(results).to include(
+          {
+            problem:  'Scenario names created by Scenario Outline are not unique',
+            location: 'path_to_file:7'
+          }
+        )
+      end
+    end
+
+    context 'with identical scenario names generated by different outlines' do
+      let(:feature_file) do
+        """
+        Feature: Feature with Conflicting Scenario Outlines
+
+          Scenario Outline: Conflicting Scenario Name <input>
+            Given something
+
+          Examples:
+            | input          |
+            | Conflict       |
+
+          Scenario Outline: Conflicting Scenario Name <input>
+            Given something else
+
+          Examples:
+            | input          |
+            | Conflict       |
+        """
+      end
+
+      let(:model) do
+        generate_feature_model(source_text: feature_file, parent_file_path: 'path_to_file').tests
+      end
+
+      it 'returns a detected problem and its location' do
+        results = model.map { |scenario| linter.lint(scenario) }.compact
+        expect(results).to include(
+          {
+            problem:  'Scenario names created by Scenario Outline are not unique',
+            location: 'path_to_file:11'
+          }
+        )
+      end
+    end
+
+    context 'with duplicate names created by scenario outline with no placeholders ' do
+      let(:feature_file) do
+        """
+        Feature: Feature with Scenario Outline
+
+          Scenario Outline: Scenario Name
+            Given something
+
+          Examples:
+            | input           |
+            | Something       |
+            | Something else  |
+        """
+      end
+
+      let(:model) do
```

Comment:
Lets DRY this let and use it within context

Reply:
Are those my fixes correct?